Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -43,8 +44,9 @@ public FreezeOnMatchCustomization(Type targetType)
         /// that will be satisfied by the frozen specimen.
         /// </param>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="targetType"/> or<paramref name="matcher"/> is null.
+        /// <paramref name="targetType"/> or <paramref name="matcher"/> is null.
         /// </exception>
+        [Obsolete("Please use FreezeOnMatchCustomization(object, IRequestSpecification) instead.")]
```

Comment:
What would happen if, instead of deprecating this overload, we change the type of the first argument from `Type` to `object`?